lib/northbound.c

```diff
@@ -549,7 +569,31 @@ int nb_candidate_edit(struct nb_config *candidate,
 		dnode = lyd_new_path(candidate->dnode, ly_native_ctx,
 				     xpath_edit, (void *)data->value, 0,
 				     LYD_PATH_OPT_UPDATE);
-		if (!dnode && ly_errno) {
+		if (dnode) {
+			/*
+			 * create dependency
+			 *
+			 * dnode returned by the lyd_new_path may be from a
+			 * different schema, so we need to update the nb_node
+			 */
+			nb_node = dnode->schema->priv;
+			if (nb_node->dep_cbs.get_dependency_xpath) {
+				nb_node->dep_cbs.get_dependency_xpath(
+					dnode, dep_xpath);
+
+				ly_errno = 0;
+				dep_dnode = lyd_new_path(candidate->dnode,
+							 ly_native_ctx,
+							 dep_xpath, NULL, 0,
+							 LYD_PATH_OPT_UPDATE);
+				if (!dep_dnode && ly_errno) {
+					flog_warn(EC_LIB_LIBYANG,
```

Comment:
nit : same error message for dnode and dep_dnode creation failure, It would be difficult to get the exact failure from logs.

Reply:
Fixed.